scripts/gen_dockerfile.py

```diff
@@ -204,12 +211,18 @@ def generate_dockerfile_command(base_image, config_file, source_dir):
     with io.open(config_file, 'r', encoding='utf8') as yaml_config_file:
         raw_config = yaml.safe_load(yaml_config_file)
 
-    # Determine complete configuration
-    app_config = get_app_config(raw_config, base_image, config_file,
-                                source_dir)
+    # Short circuit for python compat.
+    if validation_utils.get_field_value(
```

Comment:
Typo: This should be `'runtime'`, not `'runtime_id'`, right?

I'd rather see all of the "look at stuff from app.yaml" code be in the `get_app_config` function, rather than having a little bit here, and a little bit there.  E.g. in that function, before the code to look at `entrypoint`, have this check for `runtime`, and if it's value is `python-compat`, early return an AppConfig with `is_python_compat=True` and the other fields None.  `generate_files` can check that attribute.  You'll need to add an appropriate `is_python_compat` field to the AppConfig structure at line 73.